service/src/main/java/tds/assessment/services/impl/AssessmentServiceImpl.java

```diff
@@ -47,32 +48,13 @@ public AssessmentServiceImpl(AssessmentQueryRepository assessmentQueryRepository
             List<ItemProperty> itemProperties = itemQueryRepository.findActiveItemsProperties(assessmentKey);
             List<ItemConstraint> itemConstraints = assessmentQueryRepository.findItemConstraintsForAssessment(clientName,
                     assessment.getAssessmentId());
-            assessment.setItemConstraints(itemConstraints);
+            Set<Strand> strands = strandQueryRepository.findStrands(assessmentKey);
 
-            assignPropertiesToItems(items, itemProperties);
-            assignFormsAndItemsToAssessmentSegments(assessment, forms, items);
+            AssessmentAssembler.assignPropertiesToItems(items, itemProperties);
```

Comment:
I like this idea. I started doing something like this, but thought it may be less clear to return an Assessment as I'm updating by reference. I'll make the change though.